payroll/gql_mutations.py

```diff
@@ -219,6 +225,35 @@ class Input(DeletePayrollInputType):
         pass
 
 
+class MakePaymentForPayrollMutation(BaseHistoryModelDeleteMutationMixin, BaseMutation):
+    _mutation_class = "MakePaymentForPayrollMutation"
+    _mutation_module = "payroll"
+    _model = Payroll
+
+    @classmethod
+    def _validate_mutation(cls, user, **data):
+        if type(user) is AnonymousUser or not user.has_perms(
+                PayrollConfig.gql_payroll_delete_perms):
+            raise ValidationError("mutation.authentication_required")
```

Comment:
I belive you can extend this method with super() and just check if user has perms.